### PR TITLE
add forceColor command which allows forcing color on non-tty devices

### PR DESCRIPTION
**Pull request by benni012**
Hey agauniyal,
I tried my best at implementing the forceColor ostream modifier. I'm a first timer in PR's and would really appreciate if you could take a look and give me some criticism :smile:. I hope I didn't mess up too bad :sweat_smile:.

Regarding the code, I don't really like to expose the `getIword` method, but I cannot think of any other way to give access to it.

Hope this kind of helps and thanks in advance,
benni012

### CODE REVIEW

**Comment by agauniyal** [comment]
If user enters neither of `rang::control::autoColor` or `rang::control::forceColor` then what happens? should be autoColor by default, right?

**Comment by benni012** [comment]
Yes, it's `rang::autoColor` by default because the `iword` memory is initialized as 0 when `xalloc`'ed.

**Comment by agauniyal** [comment]
The code looks good, I do have a few questions though - 

 - are you writing these values(108/109) to the stream somewhere? If not, had they been 0 or 1, would it matter?
```cpp
enum class control {
	autoColor = 108,
	forceColor = 109
};
```

 - Is this sticky implementation, sticky as in if we declare `std::cout << rang::control::forceColor`, it stays on forceColor mode till we don't change it.

**Comment by benni012** [comment]
1. They are not written to the stream, they are just constants I used to handle the control commands. They could be any value. Makes way more sense to make them 0 and 1 now that I think about it. :smile:

2. The forceColor flag is stream-wide. After being activated, it will stay active in the stream until a autoColor flag is sent.

**Comment by agauniyal** [comment]
Okay let's change them to 0/1 instead and I'll merge it 😄

File: include/rang.hpp [hunk 1]
```diff
@@ -146,15 +159,33 @@ using enable = typename std::enable_if
 		std::is_same<T, rang::bgB>::value,
 		std::ostream&
 	>::type;
+template <typename T>
```
Comment (agauniyal) [inline include/rang.hpp]: an empty line before this to help readablity ;)

**Comment by agauniyal** [comment]
There's a subtle warning by clang builds too - https://travis-ci.org/agauniyal/rang/jobs/164220342#L346

**Comment by agauniyal** [comment]
I would like to thank you for taking out time to add this feature into rang library 😸

**Comment by benni012** [comment]
You're very welcome :smile:. Thank you for the warm welcome to the open source community.

**Pull request merged**